lib/LedgerSMB/Scripts/payment.pm

```diff
@@ -1115,8 +1115,11 @@ if (!$request->{exrate}) {
 # LETS GET THE DEPARTMENT INFO
 # WE HAVE TO SET $dbPayment->{department_id} in order to process
 if ($request->{department}) {
- $request->{department} =~ /^(\d+)--*/;
- $Payment->{department_id} = $1;
+    if ( $request->{department} =~ /^(\d+)--*/ ) {
+        $Payment->{department_id} = $1;
+    } else {
+        die "Error: Invalid data";
```

Comment:
Departments are defined in the `business_unit` table, with a specific `business_unit_class`. It's completely acceptable for a company to be set up without departments, in which case I'd expect the `$request->{department}` to be `undef` (not submitted).